### PR TITLE
Correctly Macro.escape :quote triplet without meta

### PR DESCRIPTION
**Pull request by kristofka**
Attempts to solve #13593 
Attempts to solve an issue that causes `Macro.escape/2` to crash when
given certain tuples.

Note that, as discussed in the reported issue, `Macro.escape({:quote, [column: 10], [:foo]})` still produces
: `{:{}, [], [:quote, [], [:foo]]}` which is arguably wrong.

### CODE REVIEW

**Comment by josevalim** [comment]
I think we should have the is_list check a bit higher. We should not even be attempting to keydelete it. We should possibly be adding is_list(Meta) to all clauses here in `do_quote`: https://github.com/elixir-lang/elixir/blob/2cc4372f1b8301d1156286b8263b53a9624fbdfa/lib/elixir/src/elixir_quote.erl#L266

**Comment by kristofka** [comment]
You are correct, this makes the intent cleaner. Unfortunately, I'm still unsure how to avoid passing something of the form `{:quote, [column: :bar], [:foo]}` . So far my attempts to restrict the match at the level of the first clause have failed.

File: lib/elixir/test/elixir/macro_test.exs [hunk 1]
```diff
@@ -38,6 +38,11 @@ defmodule MacroTest do
       assert Macro.escape({:a, {1, 2, 3}, :c}) == {:{}, [], [:a, {:{}, [], [1, 2, 3]}, :c]}
     end
 
+    test "correctly escapes tuples where the first element is :quote" do
```
Comment (josevalim) [inline lib/elixir/test/elixir/macro_test.exs:41]:
```suggestion

    # False positives
```

File: lib/elixir/test/elixir/macro_test.exs [hunk 2]
```diff
@@ -36,6 +36,10 @@ defmodule MacroTest do
       assert Macro.escape({:a}) == {:{}, [], [:a]}
       assert Macro.escape({:a, :b, :c}) == {:{}, [], [:a, :b, :c]}
       assert Macro.escape({:a, {1, 2, 3}, :c}) == {:{}, [], [:a, {:{}, [], [1, 2, 3]}, :c]}
+
+    # False positives
```
Comment (josevalim) [inline lib/elixir/test/elixir/macro_test.exs:40]:
```suggestion
      # False positives
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**